src/providers/packet/mock_tests.rs

```diff
@@ -34,3 +34,42 @@ fn test_boot_checkin() {
     mock.assert();
     r.unwrap();
 }
+
+#[test]
+fn test_packet_attributes() {
+    let metadata = r#"{
+        "id": "test-id",
+        "hostname": "test-hostname",
+        "iqn": "test-iqn",
+        "plan": "test-plan",
+        "facility": "test-facility",
+        "tags": [],
+        "ssh_keys": [],
+        "network": {
+            "interfaces": [],
+            "addresses": [],
+            "bonding": { "mode": 0 }
+        },
+        "phone_home_url": "test-url"
+    }"#;
+
+    let hostname = "test-hostname";
+    let phone_home_url = "test-url";
+    let plan = "test-plan";
+
+    let attributes = maplit::hashmap! {
+        format!("PACKET_HOSTNAME") => String::from(hostname),
+        format!("PACKET_PHONE_HOME_URL") => String::from(phone_home_url),
+        format!("PACKET_PLAN") => String::from(plan),
+    };
+
+    let _m = mockito::mock("GET", "/metadata")
+        .with_status(200)
+        .with_body(metadata)
+        .create();
+
+    let provider = packet::PacketProvider::try_new().unwrap();
+    let v = provider.attributes().unwrap();
+
+    assert_eq!(v, attributes);
+}
```

Comment:
Let's throw a `mockito::reset()` at the bottom here, and a `try_new` + `attributes().unwrap_err()` too.

Reply:
`try_new()` and `unwrap_err()` might be sufficient since in packet we are fetching the data during `try_new`, and `attributes()` would just read the struct fields.

Reply:
Ack (that behavior is quite terrible but not in scope for this PR).